exaudfclient/docs/script_options_requirments.md

```diff
@@ -0,0 +1,330 @@
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+- `feat~java-specific-script-options~1`
+
+
+### Escape Sequence Script Options Parsing
+`req~escape-sequence-script-options-parsing~1`
+
+The following rules for escape sequences at any place within a script optionValue are to be applied:
+- '\n' => <line feed> character
+- '\r' => <carriage return> character
+- '\;' => ';' character
+
+Needs: dsn
+
+Tags: V2
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+
+### Java %scriptclass Option Handling V1
+`req~java-scriptclass-option-handling-v1~1`
+
+The Java parser handler must correctly identify the first `%scriptclass` option and remove only this single instance from the script code. Any further occurrences of `%scriptclass` option shall stay in the source script code.
+The value should be handled according to the [Java specification for identifies](https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.8).
+
+Needs: dsn
+
+Tags: V1
+
+Covers:
+- `feat~java-specific-script-options~1`
+
+
+### Java %scriptclass Option Handling V2
+`req~java-scriptclass-option-handling-v2~1`
+
+The Java parser handler must correctly identify the first `%scriptclass` option and remove any additional occurrences of this option within the script code.
+The value should be handled according to the [Java specification for identifies](https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.8).
+
+Needs: dsn
+
+Tags: V2
+
+Covers:
+- `feat~java-specific-script-options~1`
+
+### Java %jar Option Handling V1
+`req~java-jar-option-handling-v1~1`
+
+The Java parser handler must find multiple %jar options. The values are to be interpreted as the Java CLASSPATH: `<file1>:<file2>:...:<filen>`.
+The Java parser handler shall split the entries by the colon character. The Java parser handler shall identify duplicated files and order the result of all `%jar` options alphabetically.
+
+Needs: dsn
+
+Tags: V1
+
+Covers:
+- `feat~java-specific-script-options~1`
+
+### Java %jar Option Handling V2
+`req~java-jar-option-handling-v2~1`
+
+The Java parser handler must find multiple %jar options. The values are to be interpreted as the Java CLASSPATH: `<file1>:<file2>:...:<filen>`.
+The Java parser handler shall split the entries by the colon character. The Java parser handler must keep duplicates. The order of the entries must not change.
+
+Needs: dsn
+
+Tags: V2
+
+Covers:
+- `feat~java-specific-script-options~1`
+
+### Java %jar Option Trailing White Space Handling
+`req~java-jar-option-trailing-white-space-handling~1`
+
+The Java parser handler must remove trailing white spaces for `%jar` option values if they are part of the escape sequence '\ '. Escape sequences at the end of a found `%jar` option of the form `\ ` must be replaced with ' '.
```

Comment:
Are you sure with trimming the spaces at the.beginning. a relative path could still have spaces, however, how would a jar land at that path in the udf

Reply:
The requirement is a bit tricky, I haven't mentioned whitespaces at the beginning because this is already covered by `req~white-spaces-script-options-parsing-v2~1`. With the legacy implementation trimming always happened. Now you can explicitely define whitespace at the beginning of an option by using the escape sequences.